src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala

```diff
@@ -27,6 +28,19 @@ class ConsignmentService(
 
   val maxLimit: Int = config.getInt("pagination.consignmentsMaxLimit")
 
+  def startUpload(startUploadInput: StartUploadInput): Future[String] = {
+    consignmentStatusRepository.getConsignmentStatus(startUploadInput.consignmentId).flatMap(status => {
+      val uploadStatus = status.find(s => s.statustype == "Upload")
+      if(uploadStatus.isDefined) {
+        throw ConsignmentStateException(s"Consignment status is ${uploadStatus.get.value}")
```

Comment:
I'm trying to think of what it would be useful to see in the error if it actually happened. What about `s"Existing consignment upload status is '${uploadStatus.get.value}', so cannot start new upload"`?

Reply:
That is more descriptive. I've changed it here and updated the tests.